src/features/wav_iterator.py

```diff
@@ -61,12 +62,23 @@ def batcher(feature_iter, batch_size=256):
         # grouped together
         try:
             batch_transposed = []
-            for dataset in list(zip(*new_batch)):
+            data_slice = list(zip(*new_batch))
+
+            # Code if we require the key to be returned
+            if return_key:
+                if type(feature_iter)!=mixer.FeatureMixer: 
```

Comment:
I can't see a good reason to split the behavior here based on the type of iterator. If there is, maybe you can explain it in a comment and split data_slice[0] and data_slice[1] into named variables to give a better idea of what's going on. 
From what I can see, the issue is that the columns being batched over need to be transposed an extra time to account for the extra level of nesting. The edits I pushed to https://github.com/pcallier/Magnolia/tree/batching-metadata do that in a more general way.